strawberryfields/apps/qchem/dynamics.py

```diff
@@ -117,3 +118,80 @@ def op(t, Ul, w, q):
         sf.ops.Interferometer(Ul) | q
 
     return op
+
+
+def sample_fock(
+    input_state: list,
+    t: float,
+    Ul: np.ndarray,
+    w: np.ndarray,
+    n_samples: int = 1,
+    loss: float = 0.0,
+    cutoff: int = 5,
+) -> list:
+
+    r"""Generate samples for simulating vibrational quantum dynamics with a Fock input state.
+
+    **Example usage:**
+    >>> input_state = [0, 2]
+    >>> t = 10.0
+    >>> Ul = np.array([[0.707106781, -0.707106781],
+    >>>                [0.707106781, 0.707106781]])
+    >>> w = np.array([3914.92, 3787.59])
+    >>> n_samples = 5
+    >>> sample_fock(input_state, t, Ul, w, n_samples = n_samples)
+    [[0, 2], [0, 2], [1, 1], [0, 2], [0, 2]]
+
+    Args:
+        input_state (list): input Fock state
+        t (float): time in femtoseconds
+        Ul (array): normal to local transformation matrix
+        w (array): normal mode frequencies :math:`\omega` in units of :math:`\mbox{cm}^{-1}`
+        n_samples (int): number of samples to be generated
+        loss (float): loss parameter denoting the fraction of generated photons that are lost
+        cutoff (int): cutoff dimension for determining the set of number states for each mode
+
+    Returns:
+        list[list[int]]: a list of samples
+    """
+    if n_samples < 1:
```

Comment:
These are great checks! I think an important one missing is verifying that the length of the input list is equal to the dimension of the unitary